frontend/src/components/eventPage/EventPage.jsx

```diff
@@ -1,48 +1,61 @@
-import React, { useState, useEffect } from "react";
-import Header from "../header/Header";
-import DisplayEventInfo from "./DisplayEventInfo";
-import DisplayEventComments from "./DisplayEventComments";
+import { useState, useEffect } from 'react'
+import Header from '../header/Header'
+import DisplayEventInfo from './DisplayEventInfo'
+import DisplayEventComments from './DisplayEventComments'
+import LoadingPage from '../loadingPage/LoadingPage'
 
 const EventPage = () => {
-  const current_link = window.location.pathname;
-  const current_eventId = current_link.split("/")[2];
-  const [event, setEvent] = useState();
-  const [comments, setComments] = useState();
-  let [changeInEvent, setchangeInEvent] = useState(true);
+    const current_link = window.location.pathname
+    const current_eventId = current_link.split('/')[2]
+    const [event, setEvent] = useState()
+    const [comments, setComments] = useState()
+    let [changeInEvent, setchangeInEvent] = useState(true)
+    const [isLoading, setIsLoading] = useState(true)
 
-  useEffect(() => {
-    if (changeInEvent) {
-      fetchEvent();
-      fetchComments();
-    }
-    setchangeInEvent(false);
-  }, [event]);
+    useEffect(() => {
+        if (changeInEvent) {
+            setIsLoading(true)
+            fetchEvent()
+            fetchComments()
+        }
+        setIsLoading(false)
+        setchangeInEvent(false)
+    }, [event])
```

Comment:
I see this logic pretty much on every module. Would you mind elaborating on what you are trying to achieve here? 
I think what we need here is simply fetchingEvents every time we load the page.

Reply:
Its because the first times it fetches an empty array, and the code breaks

Reply:
To Do:
Fix setchangeInEvent

Reply:
Done